src/erlcloud_redshift.erl

```diff
@@ -0,0 +1,279 @@
+       {value, "Value", optional_text}
+      ], Xml).
+
+extract_vpc_security_groups(Xml) ->
+    erlcloud_xml:decode(
+      [
+       {status, "Status", optional_text},
+       {vpc_security_group_id, "VpcSecurityGroupId", optional_text}
+      ], Xml).
+
+extract_cluster(Xml) ->
+    erlcloud_xml:decode(
+      [
+       {allow_version_upgrade, "AllowVersionUpgrade", optional_boolean},
+       {automated_snapshot_retention_period, "AutomatedSnapshotRetentionPeriod",
+        optional_integer},
+       {availability_zone, "AvailabilityZone", optional_text},
+       {cluster_create_time, "ClusterCreateTime", optional_text},
+       {cluster_identifier, "ClusterIdentifier", optional_text},
+       {cluster_nodes, "ClusterNodes/member",
+        {optional_map, fun extract_cluster_nodes/1}},
+       {cluster_parameter_groups, "ClusterParameterGroups/ClusterParameterGroup",
+        {optional_map, fun extract_cluster_parameter_groups/1}},
+       {cluster_public_key, "ClusterPublicKey", optional_text},
+       {cluster_revision_number, "ClusterRevisionNumber", optional_text},
+%%       {cluster_security_groups, "ClusterSecurityGroups", },  Not supported as only EC2 classic
+       {cluster_snapshot_copy_status, "ClusterSnapshotCopyStatus",
+        {single, fun extract_cluster_snapshot_copy_status/1}},
+       {cluster_status, "ClusterStatus", optional_text},
+       {cluster_subnet_group_name, "ClusterSubnetGroupName", optional_text},
+       {cluster_version, "ClusterVersion", optional_text},
+       {db_name, "DBName", optional_text},
+       {elastic_ip_status, "ElasticIpStatus",
+        {single, fun extract_elastic_ip_status/1}},
+       {encrypted, "Encrypted", optional_boolean},
+       {endpoint, "Endpoint", {single, fun extract_endpoint/1}},
+       {hsm_status, "HmsStatus", {single, fun extract_hsm_status/1}},
+       {kms_key_id, "KmsKeyId", optional_text},
+       {master_username, "MasterUsername", optional_text},
+       {modify_status, "ModifyStatus", optional_text},
+       {node_type, "NodeType", optional_text},
+       {number_of_nodes, "NumberOfNodes", optional_integer},
+       {pending_modified_values, "PendingModifiedValues",
+        {single, fun extract_pending_modified_values/1}},
+       {preferred_maintenance_window, "PreferredMaintenanceWindow",
+        optional_text},
+       {publicly_accessible, "PubliclyAccessible", optional_boolean},
+       {restore_status, "RestoreStatus", {single, fun extract_restore_status/1}},
+       {tags, "Tags/Tag", {optional_map, fun extract_tags/1}},
+       {vpc_id, "VpcId", optional_text},
+       {vpc_security_groups, "VpcSecurityGroups/VpcSecurityGroup",
+        {optional_map, fun extract_vpc_security_groups/1}}
+      ],
+      Xml).
+
+describe_all(Fun, Options, Config, Acc) ->
+    describe_all(Fun, [], Options, Config, Acc).
+
+describe_all(Fun, Args, Options, Config, Acc) when is_list(Args) ->
+    case apply(Fun, Args ++ [Options] ++ [Config]) of
```

Comment:
okay leave it like this and yes, it's time to commonize list_all in utils module